misc.go

```diff
@@ -332,13 +335,15 @@ var (
 	reWordEnd = regexp.MustCompile(`(\pL|\pN)([^\pL\pN]|$)`)
 )
 
+// TODO: Remove when bumping to Go 1.21+
```

Comment:
Just leaving a note here, I think it's fine to upgrade to the latest version of Go (do this in a separate PR). I think it is a good time now since a new version has been recently released, and it will be a while until the next one,